### ISSUE TITLE
Repeatedly enter Home menu with error

### ISSUE THREAD

**Issue by Liqianyu** [Dis90/plugin.video.discoveryplus#92]
Every time you enter the add-on, no error is reported when you enter the Home menu for the first time. However, Kodi will report an error if you return to the add-on's Home screen and enter it again.
[Kodi Log](https://paste.kodi.tv/xuvuleliba.kodi)

**Comment by Dis90** [comment]
Does other menus work, my list etc? Is it just home menu what is broken?

**Comment by Dis90** [comment]
Kodi 21 is not currently supported. Can you test with Kodi 20?

**Comment by Dis90** [comment]
I just tested with Kodi 21 and everything works. Can you login d+ in browser and then open page https://us1-prod-direct.discoveryplus.com/cms/routes/tabbed-home and save it to file and post it here so I can investigate why it fails.

**Comment by Dis90** [comment]
Sorry I forgot all parameters :D Can you give example of profile name which is not displayed?

**Comment by Dis90** [comment]
In developer tools find file called profiles. Is profiles listed like this:
```
    "attributes" : {
      "avatarName" : "2",
      "isDefault" : false,
      "isDeletable" : true,
      "pinRestricted" : true,
      "preferredLanguageTags" : [ ],
      "profileName" : "Testi"
```

**Comment by Dis90** [comment]
```
The developer tool displays the request URL as
https://us1-prod-direct.discoveryplus.com/cms/routes/tabbed-home?include=default&decorators=viewingHistory,isFavorite,playbackAllowed,badges
The browser returns the following
https://paste.kodi.tv/uhehoducuv

I also tried
https://us1-prod-direct.discoveryplus.com/cms/routes/tabbed-home?include=default
The browser returns the following
https://paste.kodi.tv/imesukujef
```

Sorry but I didn't found any reason why Home is not listed correctly for you but can you try following.
In addon.py replace this:
```
                                        next_page_path = [x['attributes']['url'] for x in routes if
                                                          x['id'] == taxonomyNode['relationships']['routes']['data'][0]['id']][0]

                                        plugin_url = plugin.url_for(list_page, next_page_path)

                                        folder_name = page['attributes'].get('pageMetadataTitle')

                                        helper.add_item(taxonomyNode['attributes']['name'], url=plugin_url)
```

With this:
```
                                        if taxonomyNode.get('relationships'):
                                            next_page_path = [x['attributes']['url'] for x in routes if
                                                              x['id'] == taxonomyNode['relationships']['routes']['data'][0]['id']][0]
    
                                            plugin_url = plugin.url_for(list_page, next_page_path)
    
                                            folder_name = page['attributes'].get('pageMetadataTitle')
    
                                            helper.add_item(taxonomyNode['attributes']['name'], url=plugin_url)
```

**Comment by Dis90** [comment]
By the way what skin you're using? Maybe profile names missing is skin issue.

**Comment by Dis90** [comment]
Okay. Can you post screenshot?

**Comment by Dis90** [comment]
Please try with Estuary skin.

**Comment by Dis90** [comment]
Works correctly for me but I try to make some changes after weekend.

**Comment by Dis90** [comment]
@Liqianyu Can you try replace this line https://github.com/Dis90/plugin.video.discoveryplus/blob/5d614929b2fee5323edae6d191ba788c8e74be4e/resources/lib/addon.py#L1503
with this:

`return helper.dialog('ok', helper.language(30006), 'No search query provided.')`

**Comment by Dis90** [comment]
@Liqianyu Could you try on another device? I am unable to reproduce the problem on two different devices. Also debug log would be nice.

**Comment by Dis90** [comment]
Yep. I try to investigate better option. `sys.exit()` works but will result errors in log.

**Comment by Dis90** [comment]
Fixed in 1.6.25+matrix.1
